src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -30,6 +30,14 @@ public class SocketTransportOptions
         /// </remarks>
         public bool WaitForDataBeforeAllocatingBuffer { get; set; } = true;
 
+        /// <summary>
+        /// Delay socket read/write operations until the transport pipe is read from or written to.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to false.
+        /// </remarks>
+        public bool DelaySocketOptions { get; set; }
```

Comment:
```suggestion
        public bool DelaySocketOperations { get; set; }
```

Reply:
In triage we decided on the following to avoid confusion with the existing `SocketTransportOptions.NoDelay` property:

```suggestion
        public bool DeferFirstOperation { get; set; }
```